code/src/test/kotlin/com/expediagroup/sdk/core/authentication/bearer/BearerAuthenticationInterceptorTest.kt

```diff
@@ -0,0 +1,361 @@
+
+        every { originalRequest.url } returns URL(requestUrl)
+        every { originalRequest.newBuilder() } returns requestBuilder
+        every { chain.request() } returns originalRequest
+
+        // Mocking authManager behavior
+        // Both calls to isTokenAboutToExpire() return true to trigger authenticate(), which throws IOException
+        every { authManager.isTokenAboutToExpire() } returnsMany listOf(true, true)
+        every { authManager.authenticate() } throws IOException("Network error")
+
+        // Act & Assert
+        val exception = assertThrows<ExpediaGroupAuthException> {
+            interceptor.intercept(chain)
+        }
+
+        assertEquals("Failed to authenticate", exception.message)
+        assertTrue(exception.cause is IOException)
+        verify(exactly = 2) { authManager.isTokenAboutToExpire() } // Initial check and inside synchronized block
+        verify(exactly = 1) { authManager.authenticate() }
+        verify(exactly = 0) { authManager.getAuthorizationHeaderValue() }
+        verify(exactly = 0) { chain.proceed(any()) }
+    }
+
+    @Test
+    fun `intercept should add correct Authorization header`() {
+        // Arrange
+        val authManager = mockk<BearerAuthenticationManager>(relaxed = true)
+        val interceptor = BearerAuthenticationInterceptor(authManager)
+        val chain = mockk<Interceptor.Chain>(relaxed = true)
+
+        val customToken = "Bearer custom_token_123"
+
+        // Mocking a non-authentication request
+        val requestBuilder = mockk<Request.Builder>(relaxed = true)
+        val originalRequest = mockk<Request>(relaxed = true)
+        val authorizedRequest = mockk<Request>(relaxed = true)
+        val requestUrl = "https://api.example.com/data"
+
+        every { originalRequest.url } returns URL(requestUrl)
+        every { originalRequest.newBuilder() } returns requestBuilder
+        every { requestBuilder.addHeader("Authorization", customToken) } returns requestBuilder
+        every { requestBuilder.build() } returns authorizedRequest
+        every { chain.request() } returns originalRequest
+        every { chain.proceed(authorizedRequest) } returns mockk()
+
+        // Mocking authManager behavior
+        every { authManager.isTokenAboutToExpire() } returns false
+        every { authManager.getAuthorizationHeaderValue() } returns customToken
+
+        // Act
+        val response = interceptor.intercept(chain)
+
+        // Assert
+        verify(exactly = 1) { authManager.isTokenAboutToExpire() }
+        verify(exactly = 1) { authManager.getAuthorizationHeaderValue() }
+        verify(exactly = 0) { authManager.authenticate() }
+        verify(exactly = 1) { requestBuilder.addHeader("Authorization", customToken) }
+        verify(exactly = 1) { chain.proceed(authorizedRequest) }
+        assertNotNull(response)
+    }
+
+    @Test
+    fun `ensureValidAuthentication should not call authenticate if token is not about to expire`() {
```

Comment:
let's avoid mentioning implementation details in the test name. Same for the next test function
```suggestion
    fun `should not call authenticate if token is not about to expire`() {
```

Reply:
Done.